### ISSUE TITLE
Documentation link anchors are not in the best place for some `autoattribute`s

### ISSUE THREAD

**Issue by wshanks** [Qiskit/qiskit#13085]
The anchors for attributes on `QuantumCircuit` are not placed in the best spot. For example, if I link to [metadata](https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.QuantumCircuit#metadata), the browser jumps to the "metadata" heading which comes from the `autoattribute` directive, but then I need to scroll back up to see the additional description of `metadata` in the docstring. One solution might be to move the `autoattribute` directives above the other description. A secondary issue is that the docstring description and attribute description are somewhat redundant. I wonder if the extra content should be moved into the attribute description so that only `autoattribute` is used or if an attribute should be inserted directly in the docstring without using `autoattribute`.

I am not sure if `QuantumCircuit` is the only class formatted this way with a mix of text and `autoattribute`s. I did not find another example with some quick scanning through the documentation.

**Comment by Mounik-tech** [comment]
try this
https://gofile.io/d/yMHabA
select the desired compiler in the installer

**Comment by jakelishman** [comment]
You're referring to the prose that explains how all the various components of `QuantumCircuit` and `qiskit.circuit` fit together - that's not _meant_ to be part of the `metadata` description at all, it's conceptually only a part of the top-level section prose above it.  It's in the same style of documentation as (say) Python's standard library or Sphinx use.

That metadata prose in particular probably could do with rebalancing so more of it's actually in the docstring rather than the prose, but for example of more what's intended, the little paragraph immediately below the `metadata` `autoattribute` is tying together the five `autoattribute`s that follow it.

It wasn't really deliberate to make the documentation split like that, it's more a function of how the documentation evolved over time - the docstring is much older, and when I wrote all the surrounding prose for the `qiskit.circuit` and `QuantumCircuit` pages, I generally didn't edit the underlying docstrings, just tied them together into one document (see #12403 for this page, e.g.).  I'm happy to have the prose rebalanced so more of it is in the actual docstrings, where appropriate.

**Comment by jakelishman** [comment]
Oh, apologies, actually I'm lying about the `metadata` docstring being older - apparently I wrote it during #12403 too.  Anyway, the point stands: it's no problem to put more of the text in the actual docstring if that ends up being a better documentation experience.  The intermediate prose is meant for tying chunks of things together - for single entries like `global_phase` and `metadata`, it probably mostly wants to be in the docstring.

**Comment by wshanks** [comment]
Yes, the solution might be just rebalancing the docstrings and the prose. The flow is not too bad, but the anchors are what is a little awkward. For example, I might want to tell a user that `data` generally should not be mutated directly. I would think that I would use https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.QuantumCircuit#data as the link to point to for that bu then if I send that link to a user I need to tell them to scroll back up.
